game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -647,31 +648,31 @@ private void setCanEvade(final Boolean s) {
   }
 
   public boolean getCanEvade() {
-    return canEvade;
+    return canEvade || isSub;
   }
 
   private void setIsFirstStrike(final Boolean s) {
     isFirstStrike = s;
   }
 
   public boolean getIsFirstStrike() {
-    return isFirstStrike;
+    return isFirstStrike || isSub || isSuicide;
```

Comment:
This is sad to see of course as options become super complex. What happens when I set 'isFirstStrike = false' and 'isSub = true' (well, it is first strike even though the option is set to off). I think that examples shows we lack flexibility, you cannot have a non-first strike sub unit or non-first strike suicide unit. 

Is the other option for us instead of the 'combo' property would be to update all maps? I'm not sure if there is really a good alternative to the code update proposed, I'm more thinking out loud and wondering what other options we might have.

Reply:
Yes, the "combo" attributes are going to cause problems.  Before this code change, if you had
```
      <option name="isFirstStrike" value="false"/>
      <option name="isSub" value="true"/>
```
then your unit would have isFirstStrike == true.
But if you had
```
      <option name="isSub" value="true"/>
      <option name="isFirstStrike" value="false"/>
```
then your unit would have isFirstStrike == false.

With this change, at least the order doesn't matter anymore.  isFirstStrike will always be true.

If you want a non-first strike sub unit, you can make that.  You just can't use `isSub`.  You have to set the atomic properties yourself.  Same with a non-first strike suicide unit.

Another option, I think, is to add to the documentation that the individual atomic properties of the combo properties can't be overridden.  If they want to have custom units, then they need to use the atomic properties instead.  The combo properties are there just to simplify the creation of basic units.

If we want atomic properties to be able to override combo properties, then we'll need some sort of flag to indicate that the atomic property is actually set.  Because setting a property to false is currently the same thing as not setting the property at all.  But I think we should focus the documentation on moving towards atomic properties only and just mark all of the combo properties as deprecated.  Then, somehow, indicate to the map maker that the property shouldn't be used.